pyspectral/rayleigh.py

```diff
@@ -262,80 +263,39 @@ def get_reflectance(self, sun_zenith, sat_zenith, azidiff, bandname, redband=Non
             wvl = self.get_effective_wavelength(bandname)
             wvl = wvl * 1000.0
 
-        rayl, wvl_coord, azid_coord, satz_sec_coord, sunz_sec_coord = self.get_reflectance_lut()
-
-        # force dask arrays
-        compute = False
-        if HAVE_DASK and not isinstance(sun_zenith, Array):
-            compute = True
-            sun_zenith = from_array(sun_zenith, chunks=sun_zenith.shape)
-            sat_zenith = from_array(sat_zenith, chunks=sat_zenith.shape)
-            azidiff = from_array(azidiff, chunks=azidiff.shape)
-            if redband is not None:
-                redband = from_array(redband, chunks=redband.shape)
-
-        sun_zenith = self._clip_angles_inside_coordinate_range(sun_zenith, sunz_sec_coord.max())
-        sunzsec = 1. / cos(deg2rad(sun_zenith))
-
-        sat_zenith = self._clip_angles_inside_coordinate_range(sat_zenith, satz_sec_coord.max())
-        satzsec = 1. / cos(deg2rad(sat_zenith))
-
-        shape = sun_zenith.shape
-
-        if not wvl_coord.min() < wvl < wvl_coord.max():
-            LOG.warning(
-                "Effective wavelength for band %s outside 400-800 nm range!",
-                str(bandname))
-            LOG.info(
-                "Set the rayleigh/aerosol reflectance contribution to zero!")
-            if HAVE_DASK:
-                chunks = sun_zenith.chunks if redband is None else redband.chunks
-                res = zeros(shape, chunks=chunks)
-                return res.compute() if compute else res
+        # if the user gave us non-dask arrays we'll use the dask
+        # version of the algorithm but return numpy arrays back
+        compute = HAVE_DASK and not isinstance(sun_zenith, Array)
+
+        rayleigh_refl = _get_wavelength_adjusted_lut_rayleigh_reflectance(self.reflectance_lut_filename, wvl)
+        if rayleigh_refl is None:
+            LOG.warning("Effective wavelength for band %s outside 400-800 nm range!",
```

Comment:
Just a flow comment here: wouldn't it be nicer if `_get_wavelength_adjusted_lut_rayleigh_reflectance` raise an error with that message "Effective wavelength for band %s outside 400-800 nm range!" that we catch here and issue a zeroed array in the except clause?

Reply:
Yes. I had tried something like that but decided against it for some reason. I'll try again after some refactoring of the tests.

Reply:
This is kind of done. Turns out the actual condition being checked is not for the 400-800 range, but whether or not the wavelength is within the range in the LUT wavelength variable. I think the higher level code can assume 400-800 though. Let me know what you think.

Reply:
Good. Yes, for the LUT we simulated reflectance in the 400 to 800 nm range.